src/hatch/utils/shells.py

```diff
@@ -3,12 +3,35 @@
 import sys
 from typing import TYPE_CHECKING
 
+from hatch.utils.fs import Path
+
 if TYPE_CHECKING:
     from collections.abc import Callable, Iterable
     from types import FrameType
+    from typing import TypeAlias
 
     from hatch.env.plugin.interface import EnvironmentInterface
-    from hatch.utils.fs import Path
+    from hatch.utils.platform import Platform
+
+
+Shell: TypeAlias = tuple[str, str]
```

Comment:
I would prefer if we could please put the return type directly on the function without a dedicated type alias.